### PR TITLE
test: pkg/db/config unit tests

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Description
test: pkg/db/config unit tests

## Checklist (Definition of Done)

- [x] All acceptance criteria specified in JIRA have been completed
- [x] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- ~~[ ] Documentation added for the feature~~
- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has been created for changes required on the client side~~

### CODE REVIEW

File: pkg/db/config_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	redactedPwd     = "<REDACTED>"
+	invalidPath     = "invalid"
+)
+
+func Test_AddFlags(t *testing.T) {
+	emptyFlagSet := &pflag.FlagSet{}
+	config := NewDatabaseConfig()
+	config.AddFlags(emptyFlagSet)
+
+	RegisterTestingT(t)
```
Comment (machi1990) [inline pkg/db/config_test.go:29]:
```suggestion
	g :⁼ NewWithT(t)
```

File: pkg/db/config_test.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+	// confirming that adding flags worked
+	flag, err := emptyFlagSet.GetString("db-ssl-certificate-file")
+
+	Expect(err).To(BeNil())
```
Comment (machi1990) [inline pkg/db/config_test.go:34]:
```suggestion
	g.Expect(err).To(BeNil())
```

File: pkg/db/config_test.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	Expect(flag).To(Equal("secrets/db.ca_cert"))
```
Comment (machi1990) [inline pkg/db/config_test.go:35]:
```suggestion
	g.Expect(flag).To(Equal("secrets/db.ca_cert"))
```

File: pkg/db/config_test.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+				config: NewDatabaseConfig(),
+			},
+			modifyFn: func(config *DatabaseConfig) {
+				config.PasswordFile = invalidPath
+			},
+			wantErr: true,
+		},
+	}
+
+	RegisterTestingT(t)
```
Comment (machi1990) [inline pkg/db/config_test.go:108]:
```suggestion
```

File: pkg/db/config_test.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (machi1990) [inline pkg/db/config_test.go:113]:
```suggestion
                       g := NewWithT(t)
			t.Parallel()
```

File: pkg/db/config_test.go [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+				config.Username = user
+				config.Password = password
+				config.Name = dbName
+				config.DatabaseCaCertFile = dbCert
+			},
+			want: fmt.Sprintf("host=%s port=%d user=%s password='%s' dbname=%s sslmode=%s sslrootcert=%s", host, port, user, password, dbName, sslModeEnabled, dbCert),
+		},
+	}
+
+	RegisterTestingT(t)
```
Comment (machi1990) [inline pkg/db/config_test.go:166]:
```suggestion
```

File: pkg/db/config_test.go [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			config := tt.fields.config
+			if tt.modifyFn != nil {
+				tt.modifyFn(config)
+			}
+			Expect(config.ConnectionString()).To(Equal(tt.want))
```
Comment (machi1990) [inline pkg/db/config_test.go:176]:
```suggestion
			g:= NewWithT(t)
			t.Parallel()
			config := tt.fields.config
			if tt.modifyFn != nil {
				tt.modifyFn(config)
			}
			g.Expect(config.ConnectionString()).To(Equal(tt.want))
```

File: pkg/db/config_test.go [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			config := tt.fields.config
+			if tt.modifyFn != nil {
+				tt.modifyFn(config)
+			}
+			Expect(config.LogSafeConnectionString()).To(Equal(tt.want))
+		})
+	}
```
Comment (machi1990) [inline pkg/db/config_test.go:236]:
```suggestion
	for _, testcase := range tests {
		tt := testcase
		t.Run(tt.name, func(t *testing.T) {
			g:= NewWithT(t)
			t.Parallel()
			config := tt.fields.config
			if tt.modifyFn != nil {
				tt.modifyFn(config)
			}
			g.Expect(config.LogSafeConnectionString()).To(Equal(tt.want))
		})
	}
```

File: pkg/db/config_test.go [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/pflag"
+
+	. "github.com/onsi/gomega"
+)
+
+var (
```
Comment (ziccardi) [inline pkg/db/config_test.go:12]: Why not using `const`?

Comment (pawelpaszki) [reply pkg/db/config_test.go:12]: good point. I can change that

**Review by machi1990 — APPROVED** [review]
lgtm

Thanks @pawelpaszki

**Pull request merged**